src/pocketmine/level/Level.php

```diff
@@ -723,6 +745,40 @@ public function doTick(int $currentTick){
 			$this->checkSleep();
 		}
 
+		if(!$this->weatherLocked){
+			--$this->thunderTime;
+			--$this->rainTime;
+			--$this->clearTime;
+			if($this->weather == self::WEATHER_NORM and ($this->rainTime <= 0 or $this->thunderTime <= 0)){
+				if(mt_rand(0, 100) > 95 or $this->thunderTime <= 0){
+					$this->setWeather(self::WEATHER_RAIN_THUNDER);
+				} else {
+					$this->setWeather(self::WEATHER_RAIN);
+				}
+			}elseif(($this->weather === self::WEATHER_RAIN and mt_rand(0, 3000) == 0) or $this->thunderTime <= 0){ //No exact wiki chance value.
+				$this->setWeather(self::WEATHER_RAIN_THUNDER);
+			}
+
+			if($this->weather = self::WEATHER_RAIN_THUNDER){
+				foreach($this->getChunks() as $chunk){
+					if(mt_rand(1, 100000) == 1){
+						$pk = new AddEntityPacket();
```

Comment:
> I can give you a 0.15 apk i have 25 different mcpe apks

Advertising illegal App sharing? Idk.. and sure i have old versions too. i am a dev.

@dktapps seems like they removed the lightning level event, or changed it.